gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc

```diff
@@ -261,6 +267,27 @@ int main(int argc, char* argv[]) {
       continue;
     }
 
+    // Get a register that we can use as the loop register.
+    std::optional<unsigned> loop_register = std::nullopt;
+    const auto& gr64_register_class =
+        reg_info.getRegClass(llvm::X86::GR64_NOREX2RegClassID);
+    for (unsigned i = 0; i < gr64_register_class.getNumRegs(); ++i) {
+      if (gr64_register_class.getRegister(i) == llvm::X86::RIP) continue;
+      if (used_registers.count(gr64_register_class.getRegister(i)) == 0) {
+        loop_register = gr64_register_class.getRegister(i);
+        break;
+      }
+    }
+
+    // If we can't find a loop register, skip writing out this basic block
+    // so that downstream tooling doesn't execute the incorrect number of
+    // iterations.
+    if (!loop_register && skip_no_loop_register) {
+      std::cerr
+          << "Skipping block due to not being able to find a loop register\n";
+      continue;
```

Comment:
How often does this happen on the ComPile data set?

Reply:
I don't have any statistics on that currently. Will hopefully collect them once this script is working at scale.

Reply:
No problem. But we should count them and then report the total number at the end.

Reply:
Good point. The script now outputs information on this at the end.